recursion/tower_of_hanoi.go

```diff
@@ -0,0 +1,21 @@
+package recursion
+
+import "fmt"
+
+// https://en.wikipedia.org/wiki/Tower_of_Hanoi
+
+func towerOfHanoi(n int, towerSource, towerDestination string, towerAuxiliary string) {
```

Comment:
```suggestion
func TowerOfHanoi(n int, towerSource, towerDestination string, towerAuxiliary string) {
```

Reply:
(About new folder dir)
recursion would be separate topic to learn?

Reply:
It is indeed a topic to learn but if you look into the repository code you'll find that their are many algorithms (for example: DepthFirstSearch) in the repository that fit the recursive topic but are not in the recursion folder.

Again wait for the maintainers to comment on this and see if they are okay with having another directory added within the repo😄

Reply:
Yeah, sounds good

Reply:
In the meantime, fix the GitHub checks that are failing. 

```bash
70 Error: func `towerOfHanoi` is unused (unused)
71   
72 Error: issues found
```

This would be fixed if you capitalise the first letter of the function definition - I've already suggested the change below!

Reply:
In the meantime, please fix the GitHub checks that are failing

```bash
70  Error: func `towerOfHanoi` is unused (unused)
71
72  Error: issues found
```

It will be fixed if you commit the changes that I have suggested at the top!